### ISSUE TITLE
Extremely long runtimes for lint command inside docker container

### ISSUE THREAD

**Issue by ealebed** [sqlfluff/sqlfluff#4964]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I faced with much longer (x10) runtime for the `sqlfluff lint` running in docker container comparing to the `sqlfluff lint` invoked from my host machine on the same set of files (550) and the same sqlfluff version.

Several docker images were tried (official one https://hub.docker.com/r/sqlfluff/sqlfluff and custom, with dbt packages inside). Docker containers were running in several ways (environments) for testing purpose - GCP CloudBuild, Ubuntu 20.04 and MacOS 13.4.1 (with M1) - and was observed the same behaviour...

### Expected Behaviour

time sqlfluff lint .  

1250.72s user 
15.41s system
595% cpu
3:32.46 total

### Observed Behaviour

time docker run -it --rm -v $PWD/dbt:/sql -v  local_sqlfluff:1 lint .  

1.54s user
2.72s system
0% cpu
39:23.59 total

### How to reproduce

You should at least set up:
1) dialect = bigquery
2) templater = dbt
3) invoke `dbt deps` command locally
4) run `sqlfluff lint` inside docker container, mounting files into container like this:

`docker run -it --rm -v $PWD/dbt:/sql -v  local_sqlfluff:1 lint .  `

### Dialect

bigquery

### Version

**sqlfluff** version 2.1.2
**sqlfluff-templater-dbt** version 2.1.2
**dbt-core** version 1.5.2 (also tried with 1.3.0)
**dbt-bigquery** version 1.5.3 (also tried with 1.3.0)

### Configuration

Config `dbt/.sqlfluffignore` looks like:
```bash
.venv/
target/
dbt_modules/
dbt_packages/
macros/
```

Config `dbt/.sqlfluff` looks like:
```bash 
[sqlfluff]
dialect = bigquery
templater = dbt
large_file_skip_byte_limit = 30000
processes = 0
max_line_length = 150

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:templater:dbt]
project_dir = ./cdw
profiles_dir = .
profile = cdw
target = dev

[sqlfluff:templater:jinja]
apply_dbt_builtins = true
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
This looks like it could be related to some of the other performance issues floating around - but I'm not sure how to debug this one. If anyone has any experience debugging docker related issues (or has access to someone who does), I'd love to seek input on where to start with this.

**Comment by barrywhart** [comment]
Is it still slow if you copy the SQL files into the container, rather than mounting the host files using `-v`? On some platforms, Docker volume mounts are very slow.
